lib/medtronic/processData.js

```diff
@@ -426,6 +464,84 @@ function buildBGRecords(records) {
   return postrecords;
 }
 
+function buildCGMRecords(events) {
+
+  var postrecords = [];
+  var start = null;
+  var recordsSinceTimestamp = null;
+
+  // we have to work through the CGM records in reverse, as sensor timestamps
+  // affect previous values and are subject to time changes
+
+  for( var i = events.length - 1 ; i >= 0; i-- ) {
+    var event = events[i];
+
+    if(event.jsDate != false) {
+      console.log('CGM event:',sundial.formatDeviceTime(event.jsDate), event.type.name, common.bytes2hex([event.head]),common.bytes2hex(event.date),common.bytes2hex(event.body),common.bytes2hex(event.descriptor) );
+
+      if(event.type.value === CBG_RECORD_TYPES.SENSOR_TIMESTAMP.value) {
+        start = event.jsDate;
+        recordsSinceTimestamp = 0;
+      }
+
+      if(event.body) {
+        var offset = event.body.length;
+        var index = event.index - offset;
+
+        while(offset >= 0) {
+
+          var isig = struct.extractShort(event.descriptor, offset * 2);
```

Comment:
`isig` doesn't seem to really be the isig, but a composite structure. It would be great if we could have a comment that links to where the structure is defined.
My rationale for this is that as a new developer, I can see that the code is doing something, but I have no way of validating if it's doing it correctly.